File: C++/53.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int n = nums.size();
        int res = 0;
        int max = INT_MIN;
        for(int i = 0; i < n; i++){
            res += nums[i];
            if (res > max) max = res;
            if (res < 0) res = 0;
        }
        return max;
    }
};

File: Python/53.py
# Initial thoughts
    # Any positive prefix adds to sum, so just accumulate sum while positive
    # Compare max at each step
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        current_sum = 0
        max_sum = nums[0]
        for i in nums:
            if current_sum < 0:
                current_sum = 0
            current_sum += i
            max_sum = max(current_sum, max_sum)

        return max_sum